tests/test-sdi-progress-dock.c

```diff
@@ -0,0 +1,154 @@
+#include "../src/sdi-progress-dock.h"
+#include "gtk/gtk.h"
+SdiProgressDock *progress_dock = NULL;
+
+static gboolean updating_value = FALSE;
+static gboolean progress_visible_value = FALSE;
+static gdouble progress_value = -1;
+static gchar *expected_program = NULL;
+
+enum CHANGES_LIST {
+  CHANGES_PROGRESS = 1,
+  CHANGES_PROGRESS_VISIBLE = 2,
+  CHANGES_UPDATING = 4
+} current_changes = 0;
+
+static void set_progress_bar(gchar *desktop_file, guint done_tasks,
+                             guint total_tasks, gboolean all_done) {
+  current_changes = 0;
+  g_autoptr(GStrvBuilder) desktop_files_builder = g_strv_builder_new();
+  g_strv_builder_add(desktop_files_builder, desktop_file);
+  g_auto(GStrv) desktop_files = g_strv_builder_end(desktop_files_builder);
+  sdi_progress_dock_update_progress(progress_dock, "snap name", desktop_files,
+                                    "a description", done_tasks, total_tasks,
+                                    all_done);
+}
+
+static void timeoutCB(guint *timeout_id) {
+  g_assert_cmpint(*timeout_id, !=, 0);
+  *timeout_id = 0;
+}
+
+static void wait_for_events(guint timeout, guint changes) {
+  GMainContext *context = g_main_context_default();
+  guint timeout_id =
+      g_timeout_add_once(timeout, (GSourceOnceFunc)timeoutCB, NULL);
+  do {
+    g_main_context_iteration(context, TRUE);
+  } while ((timeout_id != 0) && (current_changes & changes) != changes);
+  if (timeout_id != 0) {
+    g_clear_handle_id(&timeout_id, g_source_remove);
+  }
```

Comment:
No need for the if, that's the point of the `g_clear_*` thingy :)

```suggestion
  g_clear_handle_id(&timeout_id, g_source_remove);
```

Reply:
Agh! Of course! 😅